src/libponyc/expr/reference.c

```diff
@@ -675,7 +675,7 @@ bool expr_reference(pass_opt_t* opt, ast_t** astp)
 
       ast_t* type = ast_type(def);
 
-      if(is_typecheck_error(type))
+      if(is_typecheck_error(type) && (errors_get_count(opt->check.errors) > 0))
```

Comment:
Is it really the best thing to do? In a code like this

```pony
actor Main
  new create(env: Env) =>
    let x = x.create()
    let y = y.create()
```

only the error for `x` is reported. Wouldn't it be better to remove that line entierely?